File: app/src/main/java/cz/johrusk/showsmscode/sched/JobCreator.java
package cz.johrusk.showsmscode.sched;

/**
 * Class which creates scheduled job
 *
 * @author Josef Hruska (pepa.hruska@gmail.com)
 */
public class JobCreator implements com.evernote.android.job.JobCreator {


    @Override
    public com.evernote.android.job.Job create(String tag) {
        switch (tag) {

            case UpdateJob.TAG:
                return  new UpdateJob();
            case UpdateJob.TAG_WEEKLY:
                return new UpdateJob();
            default:
                return null;
        }
    }
}

File: app/src/main/java/cz/johrusk/showsmscode/sched/JobCreator.kt
package cz.johrusk.showsmscode.sched

/**
 * Class which creates scheduled job

 * @author Josef Hruska (pepa.hruska@gmail.com)
 */
class JobCreator : com.evernote.android.job.JobCreator {


    override fun create(tag: String): com.evernote.android.job.Job? {
        when (tag) {
            UpdateJob.TAG -> return UpdateJob()
            UpdateJob.TAG_ONSTART -> return UpdateJob()
            else -> return null
        }
    }
}